src/vcpkg-test/configparser.cpp

```diff
@@ -40,23 +60,35 @@ TEST_CASE ("BinaryConfigParser files provider", "[binaryconfigparser]")
     }
     {
         auto parsed = create_binary_providers_from_configs_pure("files," ABSOLUTE_PATH, {});
-        REQUIRE(parsed.has_value());
+        auto state = parsed.value_or_exit(VCPKG_LINE_INFO);
```

Comment:
Correct me if I'm wrong (I don't know much about how `catch2` works) but won't this terminate the test program in case of failure? 

I think it's ok to leave the `REQUIRE(parsed.has_value());` above and skip only this test case on failure.

Reply:
Yes, but IMO that is OK. We use value_or_default all over the product code in such cases where not having a value would be nonsense, I don't see why test code should be any different.